web-access-app/src/main/java/edu/unc/lib/boxc/web/access/controllers/FullRecordController.java

```diff
@@ -98,17 +102,33 @@
     @Autowired
     private RepositoryObjectLoader repositoryObjectLoader;
 
-    @RequestMapping(value = "/{pid}", method = RequestMethod.GET)
-    public String handleRequest(@PathVariable("pid") String pid, Model model, HttpServletRequest request) {
-        return getFullRecord(pid, model, request);
+    @GetMapping("/{pid}")
+    public ModelAndView handleRequest(@PathVariable("pid") String pid, Model model, HttpServletRequest request) {
```

Comment:
how does this go from being a string to a ModelAndView? What is ModelAndView and how does it work?

Reply:
Its a wrapper for passing along the view routing and model information for a request. The Spring MVC framework is executing these controller methods and it knows how to receive quite a few object types as the return type from controller methods, including `String`s and `ModelAndView`. I switched return types here because it allows me to set a different status code some of the time. There are some other ways to do that, but this seemed preferable.